### PR TITLE
fix core for getservbyname in multi-thread for mysql

### PR DESCRIPTION
**Pull request by qwdarrenyin**
fix core for getservbyname in multi-thread for mysql

### CODE REVIEW

File: libchannelserver/ChannelRPCServer.cpp [hunk 1]
```diff
@@ -594,6 +594,13 @@ void dev::ChannelRPCServer::asyncPushChannelMessageHandler(
     {
         CHANNEL_LOG(DEBUG) << "receive node request" << LOG_KV("topic", toTopic)
                            << LOG_KV("content", content);
+
+        if (getSessionByTopic(toTopic).empty())
```
[the remark this answers is not in the document]
Comment (qwdarrenyin) [reply libchannelserver/ChannelRPCServer.cpp]: there is no core, just to let there no error log

File: cmake/ProjectLibzdb.cmake [hunk 2]
```diff
@@ -1,29 +1,27 @@
 include(ExternalProject)
 
-if (APPLE)
-    set(MYSQL_CLIENT_URL https://cdn.mysql.com/archives/mysql-connector-c/mysql-connector-c-6.1.11-macos10.12-x86_64.tar.gz)
-    set(MYSQL_CLIENT_SHA256 c97d76936c6caf063778395e7ca15862770a1ab77c1731269408a8d5c0eb4b93)
-else()
-    set(MYSQL_CLIENT_URL https://cdn.mysql.com/archives/mysql-connector-c/mysql-connector-c-6.1.11-linux-glibc2.12-x86_64.tar.gz)
-    set(MYSQL_CLIENT_SHA256 149102915ea1f1144edb0de399c3392a55773448f96b150ec1568f700c00c929)
-endif()
+set(MYSQL_CONNECTOR_URL https://downloads.mysql.com/archives/get/file/mysql-connector-c-6.1.11-src.tar.gz)
```
Comment (bxq2011hust) [inline cmake/ProjectLibzdb.cmake]: unnecessary

File: libchannelserver/ChannelRPCServer.cpp [hunk 3]
```diff
@@ -602,9 +609,14 @@ void dev::ChannelRPCServer::asyncPushChannelMessageHandler(
         CHANNEL_LOG(DEBUG) << "async push channel message" << LOG_KV("seq", channelMessage->seq())
                            << LOG_KV("type", channelMessage->type())
                            << LOG_KV("datalen", channelMessage->length());
+
         asyncPushChannelMessage(toTopic, channelMessage,
-            std::function<void(dev::channel::ChannelException, dev::channel::Message::Ptr,
-                dev::channel::ChannelSession::Ptr)>());
+            [channelMessage](dev::channel::ChannelException e, dev::channel::Message::Ptr,
```
Comment (bxq2011hust) [inline libchannelserver/ChannelRPCServer.cpp]: add if(m_callback) to avoid another one meet the same problem

Comment (JimmyShi22) [reply libchannelserver/ChannelRPCServer.cpp]: Use LOG_DESC()  to add pure comments, same as other comments below.

**Pull request merged**